modules/foreman_compute_profile.py

```diff
@@ -0,0 +1,129 @@
+# (c) Philipp Joos 2017
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: foreman_compute_profile
+short_description: Manage Foreman Compute Profiles using Foreman API v2
+description:
+- Create and delete Foreman Compute Profiles using Foreman API v2
+options:
+  name:
+    description: compute profile name
+    required: true
+  updated_name:
+    description: new compute profile name
+    required: false
+  server_url:
+    description: foreman url
+    required: true
+  username:
+    description: foreman username
+    required: true
+  password:
+    description: foreman user password
+    required: true
+  verify_ssl:
+    description: verify ssl connection when communicating with foreman
+    default: true
+  state:
+    description: compute profile presence
+    default: present
+    choices: ["present", "absent", "latest"]
+notes:
+- Requires nailgun
+version_added: "2.0"
+author: "Philipp Joos (@philippj)"
+'''
+
+EXAMPLES = '''
+- name: compute profile
+  foreman_compute_profile:
+    name: example_compute_profile
+    server_url: foreman.example.com
+    username: admin
+    password: secret
+    verify_ssl: false
+    state: present
+'''
+
+RETURN = ''' # '''
+
+try:
+    from nailgun.config import ServerConfig
+    import nailgun.entity_mixins
+    import nailgun.entities
+    import nailgun.entity_fields
+    HAS_NAILGUN_PACKAGE = True
+
+except:
+    HAS_NAILGUN_PACKAGE = False
+
+from ansible.module_utils.basic import *
+import ansible.module_utils.ansible_nailgun_cement as cement
+
+
+def main(module):
+    if not HAS_NAILGUN_PACKAGE:
+        module.fail_json(
+            msg="""Missing required nailgun module (check docs or install)
+            with: pip install nailgun"""
+        )
+
+    name = module.params.get('name')
+    updated_name = module.params.get('updated_name')
+    state = module.params.get('state')
+
+    cement.create_server(
+        server_url=module.params.get('server_url'),
+        auth=(module.params.get('username'), module.params.get('password')),
+        verify_ssl=module.params.get('verify_ssl'),
+    )
+
+    cement.ping_server(module)
+
+    data = {
+        'name': name
+    }
+
+    compute_profile = cement.find_compute_profile(module, name=data.get('name'), failsafe=True)
+
+    if state == 'latest':
+        data['name'] = updated_name
```

Comment:
I must say I don't like that, but I am also not aware of a better way to rename stuff in Ansible

@akofink ideas?

Reply:
Wow, not sure how I missed this. Anyway, yeah, I don't like it. I don't see why we can't just have present/absent. If you want to rename something, remove it and add it back.

Reply:
Deleting and recreating means, that we loose all database references to it. This might no be, what is expected for renaming let's say 'small' to 'tiny'.

Reply:
Good point. It's fine with me to keep latest in the module.

Reply:
Yeah, sorry, little late. I realized, having present and an "updated name" is enough information to know, that the user wanted it changed.